portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -64,8 +67,10 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      changePage(request);
```

Comment:
As mentioned earlier, if the page is part of the get request, you don't need to do any page handling in the post.